lib/archethic/contracts.ex

```diff
@@ -165,6 +167,37 @@ defmodule Archethic.Contracts do
     end
   end
 
+  def is_valid_contract?(
+        prev_tx = %Transaction{data: %TransactionData{code: code}},
+        next_tx = %Transaction{},
+        date = %DateTime{}
+      ) do
+    case Interpreter.parse(code) do
+      {:ok,
+       %Contract{
+         triggers: triggers,
+         conditions: %{inherit: inherit_conditions}
```

Comment:
also you should check the transaction's condition instead of the inherit's condition.